src/NLog/Config/LoggingRuleLevelFilter.cs

```diff
@@ -0,0 +1,251 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.Config
+{
+    using System;
+    using System.Collections.Generic;
+    using NLog.Common;
+    using NLog.Internal;
+    using NLog.Layouts;
+
+    internal interface ILoggingRuleLevelFilter
+    {
+        /// <summary>
+        /// Level enabled flags for each LogLevel ordinal
+        /// </summary>
+        bool[] LogLevels { get; }
+        /// <summary>
+        /// Reconfigure existing level configuration with the specified config
+        /// </summary>
+        /// <returns>New filter</returns>
+        ILoggingRuleLevelFilter SetLoggingLevels(LogLevel minLevel, LogLevel maxLevel, bool enable);
+    }
+
+    /// <summary>
+    /// Default filtering with static level config
+    /// </summary>
+    class LoggingRuleLevelFilter : ILoggingRuleLevelFilter
+    {
+        public static readonly LoggingRuleLevelFilter Off = new LoggingRuleLevelFilter();
+        public bool[] LogLevels { get; }
+
+        public LoggingRuleLevelFilter(bool[] logLevels = null)
+        {
+            LogLevels = new bool[LogLevel.MaxLevel.Ordinal + 1];
+            if (logLevels != null)
+            {
+                for (int i = 0; i < Math.Min(logLevels.Length, LogLevels.Length); ++i)
+                    LogLevels[i] = logLevels[i];
+            }
+        }
+
+        public ILoggingRuleLevelFilter SetLoggingLevels(LogLevel minLevel, LogLevel maxLevel, bool enable)
+        {
+            if (!ReferenceEquals(LogLevels, Off.LogLevels))
+            {
+                for (int i = minLevel.Ordinal; i <= maxLevel.Ordinal; ++i)
+                    LogLevels[i] = enable;
+                return this;
+            }
+
+            return new LoggingRuleLevelFilter().SetLoggingLevels(minLevel, maxLevel, enable);
+        }
+    }
+
+    /// <summary>
+    /// Dynamic filtering with a positive list of enabled levels
+    /// </summary>
+    class DynamicLogLevelFilter : ILoggingRuleLevelFilter
+    {
+        private static readonly char[] _levelFilerSplitter = { ',' };
+        readonly LoggingRule _loggingRule;
+        readonly SimpleLayout _levelFilter;
+        private KeyValuePair<string, bool[]> _activeFilter;
+
+        public bool[] LogLevels => GenerateLogLevels();
+
+        public DynamicLogLevelFilter(LoggingRule loggingRule, SimpleLayout levelFilter)
+        {
+            _loggingRule = loggingRule;
+            _levelFilter = levelFilter;
+            _activeFilter = new KeyValuePair<string, bool[]>(string.Empty, LoggingRuleLevelFilter.Off.LogLevels);
+        }
+
+        public ILoggingRuleLevelFilter SetLoggingLevels(LogLevel minLevel, LogLevel maxLevel, bool enable)
+        {
+            return new LoggingRuleLevelFilter(LogLevels).SetLoggingLevels(minLevel, maxLevel, enable);
+        }
+
+        private bool[] GenerateLogLevels()
```

Comment:
please split this method in smaller ones, maybe some code could be deduplicated?

Reply:
Fixed